invenio_index_migrator/api/migration.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Index syncing API."""
+
+from __future__ import absolute_import, print_function
+
+import json
+import warnings
+from datetime import datetime
+
+from elasticsearch import VERSION as ES_VERSION
+from elasticsearch.exceptions import NotFoundError
+from flask import current_app
+from invenio_search.api import RecordsSearch
+from invenio_search.proxies import current_search, current_search_client
+from invenio_search.utils import build_alias_name, build_index_name, \
+    prefix_index
+from six import string_types
+from werkzeug.utils import cached_property
+
+from ..indexer import SYNC_INDEXER_MQ_QUEUE, MigrationIndexer
+from ..proxies import current_index_migrator
+from ..tasks import run_sync_job
+from ..utils import ESClient, State, extract_doctype_from_mapping, \
+    get_queue_size, obj_or_import_string
+
+
+def ensure_valid_config(f):
+    """Decorate to ensure that all config parametersare valid."""
+    def inner(self, name, **config):
+        missing = [p for p in self.REQUIRED_PARAMS if p not in config]
```

Comment:
I saw that `Migration` defines `REQUIRED_PARAMS = ('src_es_client', 'jobs', 'strategy')`, but in the example configuration of `config.py` the `strategy` param is contained in the `src_es_client` dictionary. I fear that for this reason the check above won't be able to find the `strategy` param.

Reply:
I saw the in `conftest.py` the `strategy` field is top-level, so I will update the example.